src/Core/Console.php

```diff
@@ -15,6 +16,14 @@ class Console extends \Asika\SimpleConsole\Console
 	protected $helpOptions = [];
 	protected $customHelpOptions = ['h', 'help', '?'];
 
+	private $dice;
+
+	public function __construct(Dice $dice, $argv = null)
```

Comment:
I thought it was an anti-pattern to pass the Dependency Injection Container as a class parameter?

Reply:
You're right, but I've to start at a place.
It seems for me that there are three central "controllers", which are used to determine the initial class we need:
- `Friendica\Core\Console` => Determine the `Friendica\Console\*` class
- `Friendica\Core\Worker` => Determine the `Friendica\Worker\*` class
- `Friendica\App\Router` => Determine the `Friendica\Module\*` class

Every other classes "could" be loaded through the determined class in a later stage.

Reply:
So, reading further I found out that it's okay to pass the DI container to a class that will act as a Composition Root like I believe the `Core\Console` class is.